src/app/components/PasswordField/index.tsx

```diff
@@ -2,39 +2,45 @@ import { Box, FormField, Button, TextInput, Tip } from 'grommet'
 import { View, Hide } from 'grommet-icons'
 import * as React from 'react'
 
-interface Props {
-  placeholder: string
+interface Props<TFormValue> {
+  name: Extract<keyof TFormValue, string>
+  label?: string
+  placeholder?: string
   inputElementId: string
 
   autoComplete: 'on' | 'off' | 'new-password' | 'current-password'
-  value: string
-  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
-  error: string | false
+  value?: string
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
```

Comment:
It looks like these type changes does not belong to this PR. I don't see any use cases for optional `value` and `onChange` or 'TFormValue' here.

Reply:
The new recommended usage is with `name` and `validate`, but `value`, `onChange`, `error` still come useful. And keeping them is consistent with grommet's FormField.

I added some tests to validate both uses